.github/workflows/sda-pipeline.yml

```diff
@@ -0,0 +1,167 @@
+name: sda-pipeline deployment
+
+on: [push,pull_request]
+
+env:
+  posix_svc_list: 'backup doa finalize inbox ingest intercept mapper verify'
+  s3_svc_list: 'auth backup doa finalize inbox ingest intercept mapper verify'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        test: [sftp-inbox, s3-inbox]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install kube dependencies
+      run: bash ./dev_tools/scripts/install-kube-deps.sh
+    - name: Initialise k3d
+      run: bash ./dev_tools/scripts/init-k3d.sh
+    - name: Wait for k3d to become ready
+      run: until kubectl -n kube-system get pods -lk8s-app=metrics-server -o jsonpath='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status}' | grep "Ready=True"; do echo "waiting for k3d to become ready"; sleep 10; done
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Install sda dependencies
+      run: bash ./dev_tools/scripts/install-sda-deps.sh
+    - name: Set up services configuration
+      run: bash ./dev_tools/scripts/svc-setup.sh
+    - name: Deploy SDA database
+      run: bash ./dev_tools/scripts/deploy-db.sh
+    - name: Wait for database to become ready
+      run: |
+        RETRY_TIMES=0
+        until kubectl get pods -lrole=database -o jsonpath='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status}' | grep "Ready=True";
+        do echo "waiting for database to become ready";
+        RETRY_TIMES=$((RETRY_TIMES+1));
+        if [ $RETRY_TIMES -eq 10 ]; then exit 1; fi
+        sleep 10;
+        done
+    - name: debug DB
+      if: failure()
+      run: kubectl describe pod -l role=database
+    - name: log DB
+      if: failure()
+      run: kubectl logs postgres-sda-db-0
+    - name: Deploy mock oidc server
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/deploy-oidc.sh
+    - name: Deploy minio
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/deploy-minio.sh
+    - name: Wait for minio to become ready
+      if: matrix.test == 's3-inbox'
+      run: |
+        RETRY_TIMES=0
+        until kubectl get pods -lapp=minio -o jsonpath='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status}' | grep "Ready=True";
+        do echo "waiting for minio to become ready"
+        RETRY_TIMES=$((RETRY_TIMES+1));
+        if [ $RETRY_TIMES -eq 10 ]; then exit 1; fi
+        sleep 10;
+        done
+    - name: debug minio
+      if: failure() && matrix.test == 's3-inbox'
+      run: kubectl describe pod -l app=minio
+    - name: log minio
+      if: failure() && matrix.test == 's3-inbox'
+      run: kubectl logs $(kubectl get pods | grep minio | awk '{print $1}')
```

Comment:
```suggestion
      run: kubectl logs $(kubectl get pod -o name -l app=minio)
```

Reply:
Yes. this works if the resources have the right labels. But perhaps we should not trust this?